girder/api/v1/resource.py

```diff
@@ -139,6 +140,96 @@ def _getResourceModel(self, kind, funcName=None):
             raise RestException('Invalid resources format.')
         return model
 
+    def _lookupToken(self, token, parentType, parent):
+        """
+        Find a particular child resource by name or throw an exception.
+        :param token: the name of the child resource to find
+        :param parentType: the type of the parent to search
+        :param parent: the parent resource
+        :returns: the child resource
+        """
+
+        filterObject = {'name': token, 'parentId': parent['_id']}
+
+        # look for a folder
+        candidateChild = self.model('folder').findOne(filterObject)
+        if candidateChild is not None:
+            return candidateChild, 'folder'
+
+        # if folder not found, look for an item
+        filterObject['folderId'] = filterObject.pop('parentId')
+        candidateChild = self.model('item').findOne(filterObject)
+        if candidateChild is not None:
+            return candidateChild, 'item'
+
+        # if item not found, look for a file
+        filterObject['itemId'] = filterObject.pop('folderId')
+        candidateChild = self.model('file').findOne(filterObject)
+        if candidateChild is not None:
+            return candidateChild, 'file'
+
+        # if no folder, item, or file matches, give up
+        raise RestException('Child resource not found: {}({})->{}'.format(
+            parentType, parent.get('name', parent.get('_id')), token))
+
+    def _lookupPath(self, path, user=None):
+        """
+        Find a particular resource by path or throw an exception.  The given
+        user must have read access to all resources featured in the path.
+        :param path: the path of the object to find
+        :param user: the user to search as
+        :returns: the resource.
+        """
+
+        pathArray = [token for token in path.split('/') if token]
+        model = pathArray[0]
+
+        if user is None:
+            user = self.getCurrentUser()
```

Comment:
Let's remove the above block and simply pass the user in rather than provide a default. `None` is actually a valid user, and since this is a private method it won't be a big deal to force passing the user.